src/org/sipdroid/sipua/SipdroidEngine.java

```diff
@@ -52,52 +53,57 @@
 import org.zoolu.sip.provider.SipStack;
 import org.zoolu.tools.Timer;
 
-import android.content.Context;
-import android.content.SharedPreferences.Editor;
-import android.net.Uri;
-import android.net.wifi.WifiManager;
-import android.os.Build;
-import android.os.PowerManager;
-import android.os.SystemClock;
-import android.preference.PreferenceManager;
-import android.util.Log;
-
-import org.omnidial.harvest.DialCandidate;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.KeyStore;
+import java.util.List;
 
 public class SipdroidEngine implements RegisterAgentListener {
 
 	//public static final int LINES = 2;
 	public int pref;
-	
+
 	public static final int UNINITIALIZED = 0x0;
 	public static final int INITIALIZED = 0x2;
-	
-	/** User Agent */
+
+	/**
+	 * User Agent
+	 */
 	public UserAgent[] uas;
 	public UserAgent ua;
 
-	/** Register Agent */
+	/**
+	 * Register Agent
+	 */
 	public RegisterAgent[] ras;
-	
-	/** Messaging */
+
+	/**
+	 * Publish Agent
+	 */
+	public PublishAgent[] pas;
+	/**
+	 * Messaging
+	 */
 	public MessageAgent[] mas;
 
 	private KeepAliveSip[] kas;
-	
-	/** UserAgentProfile */
+
+	/**
+	 * UserAgentProfile
+	 */
 	public UserAgentProfile[] user_profiles;
 
 	public SipProvider[] sip_providers;
-	
-	public static PowerManager.WakeLock[] wl,pwl;

```

Comment:
please don't fix formatting like this, although you can make such fixes on a different branch if you like, then I can study them separately